src/databricks/labs/ucx/source_code/linters/dfsa.py

```diff
@@ -0,0 +1,176 @@
+from dataclasses import dataclass
+import logging
+from abc import ABC
+from collections.abc import Iterable
+
+from astroid import Call, Const, InferenceError, NodeNG  # type: ignore
+from sqlglot import Expression
+from sqlglot.expressions import Table
+
+from databricks.labs.ucx.source_code.base import (
+    Advice,
+    Deprecation,
+    CurrentSessionState,
+    PythonLinter,
+    SqlLinter,
+)
+from databricks.labs.ucx.source_code.python.python_ast import Tree, TreeVisitor
+from databricks.labs.ucx.source_code.python.python_infer import InferredValue
+
+logger = logging.getLogger(__name__)
+
+
+class DFSAPattern(ABC):
+
+    def __init__(self, prefix: str, allowed_roots: list[str]):
+        self._prefix = prefix
+        self._allowed_roots = allowed_roots
+
+    def matches(self, value: str) -> bool:
+        return value.startswith(self._prefix) and not self._matches_allowed_root(value)
+
+    def _matches_allowed_root(self, value: str):
+        return any(value.startswith(f"{self._prefix}/{root}") for root in self._allowed_roots)
+
+
+class RootPattern(DFSAPattern):
+
+    def _matches_allowed_root(self, value: str):
+        return any(value.startswith(f"/{root}") for root in self._allowed_roots)
+
+
+# the below aims to implement https://docs.databricks.com/en/files/index.html
+DFSA_PATTERNS = [
+    DFSAPattern("dbfs:/", []),
+    DFSAPattern("file:/", ["Workspace/", "tmp/"]),
```

Comment:
```suggestion
    DFSAPattern("file:/", ["Workspace/"]),
```
tmp is irrelevant here

Reply:
done